crates/api/src/wire.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+
+package kitsune2.wire;
+
+// A Kitsune2 wire protocol message.
+//
+// This is the top-level encoding
+// that will be transferred between Kitsune2 peers. Most communications
+// between peers to make Kitsune2 actually function will be encoded
+// separately inside the payload of TY_MODULE type messages.
+message K2Proto {
+  // Enumeration of the types of messages that can be sent between peers.
+  //
+  // We are using this enum field to distinguish between top-level messages,
+  // rather than protobuf's oneof because of the downsides of upgrading oneofs.
```

Comment:
What are the downsides of `oneof`'s?

Reply:
[Mostly that they are very fiddly to update if you can at all](https://protobuf.dev/programming-guides/proto3/#backward)